openadapt/record.py

```diff
@@ -721,27 +774,41 @@ def read_window_events(
             started = True
 
         if window_data["title"] != prev_window_data.get("title") or window_data[
-            "window_id"
-        ] != prev_window_data.get("window_id"):
+            "handle"
+        ] != prev_window_data.get("handle"):
             # TODO: fix exception sometimes triggered by the next line on win32:
             #   File "\Python39\lib\threading.py" line 917, in run
             #   File "...\openadapt\record.py", line 277, in read window events
             #   File "...\env\lib\site-packages\loguru\logger.py" line 1977, in info
             #   File "...\env\lib\site-packages\loguru\_logger.py", line 1964, in _log
             #       for handler in core.handlers.values):
             #   RuntimeError: dictionary changed size during iteration
-            _window_data = window_data
+            _window_data = window_data.copy()
             _window_data.pop("state")
             logger.info(f"{_window_data=}")
+            counter = 0
         if window_data != prev_window_data:
             logger.debug("Queuing window event for writing")
-            event_q.put(
-                Event(
-                    utils.get_timestamp(),
-                    "window",
-                    window_data,
-                )
+            counter += 1
+            window_data.update({"counter": counter})
+            _window_data = window_data.copy()
+            _window_data.pop("state", None)
+            window_event = Event(
+                utils.get_timestamp(),
+                "window",
+                _window_data,
+            )
+            event_q.put(window_event)
+            a11y_event = Event(
+                utils.get_timestamp(),
+                "a11y",
+                window_data,
+            )
+            window_event_by_id[(window_data["handle"], window_data["counter"])] = (
```

Comment:
How about defining the tuple on a separate line? e.g
```
window_tup = (window_data["handle"], window_data["counter"])
windwo_event_by_id[window_tup] = ...
```

Reply:
Why do we need to place the event in this dict, as well as put it in the event queue?